scripts/check_repo_stats.py

```diff
@@ -0,0 +1,175 @@
+
+#### End Requirement Callables ####
+
+
+def check_requirements(
+    repo: str, requirements: list[Callable[[dict, Optional[str]], bool]], reqs: list[str], metadata: Optional[dict] = None
+) -> bool:
+    """Checks if Github repository meets requirements
+
+    Args:
+        repo (str): Github repository in repo_owner/repo_name format
+        requirements (list[callable]): List of requirement callables that check if repo meets requirement
+        reqs (list[str]): List of values to check against for each callable
+            - each req will be called with the callable of the same index in requirements
+
+    Returns:
+        bool: True if repo meets requirements, False otherwise
+    """
+
+    repo_query = repo.split("/")
+
+    # Get repo data
+    gql_format = """
+    query {
+        rateLimit {
+            cost
+            remaining
+            resetAt
+        }
+        repository(name:"%s", owner:"%s"){
+            id
+            owner {
+                login
+            }
+            name
+            url
+            isArchived
+            primaryLanguage {
+                name
+            }
+            pushedAt
+            stargazerCount
+            object(expression: "HEAD:") {
+                ... on Tree {
+                    entries {
+                        name
+                        type
+                    }
+                }
+            }
+        }
+    }
+    """
+    # Example of format of metadata:
+    # {
+    #   'data': {
+    #       'rateLimit': {'cost': <cost>, 'remaining': <remaining>, 'resetAt': <ISO-8601 datetime>},
+    #       'repository': {'primaryLanguage': <language>, 'pushedAt': <ISO-8601 datetime>, 'stargazerCount': <num_stars>}
+    #   }
+    # }
+    data = metadata
+    if metadata is None:  # Query if metadata is not provided
+        metadata = get_graphql_data(gql_format % (repo_query[1], repo_query[0]))
+        data = metadata["data"]["repository"]
+    if "errors" in metadata:
+        sys.exit(f"Fetching repo metadata error: {metadata['errors']}")
+
+    print(f"{repo} metadata: {metadata}")
+
+    # If repo is archived, automatic fail
+    if data["isArchived"]:
+        print("Repo is archived")
+        return False
+
+    # Check requirements
+    for i in range(len(requirements)):
+        if requirements[i] == req_fuzzers and not requirements[i](data):
+            print(f"Error with req {requirements[i].__name__}")
+            return False
+        elif requirements[i] != req_fuzzers and not requirements[i](data, reqs[i]):
+            print(
+                f"Error with req {requirements[i].__name__} with requirement {reqs[i]}"
+            )
+            return False
+    return True
+
+
+# Pass checks_list and reqs with this : --checks_list='<list>' --reqs='<list>'
+# Ex. --reqs='["0", "2020-1-1"]'template
+# If checking Rust fuzz path, put null in place of where the req should be in the reqs list
+def main(
+    repo_id_list: str = "ethanbwang/test",
```

Comment:
What default should I provide it with? Since that's just a repo name.

Reply:
> Try not to hard code path `"ethanbwang"`

"ethanbwang/test" is actually the repo id for https://github.com/ethanbwang/test instead of a path @DataCorrupted